src/test/java/io/nats/compatibility/ObjectStoreCommand.java

```diff
@@ -0,0 +1,280 @@
+            ObjectMeta meta = ObjectMeta.builder(objectName).description(description).build();
+            os.updateMeta(object, meta);
+            respond();
+        }
+        catch (Exception e) {
+            handleException(e);
+        }
+    }
+
+    private void doGetLink() {
+        try {
+            ObjectStore os = nc.objectStore(bucket);
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            _respondDigest(os.get(object, baos));
+        }
+        catch (Exception e) {
+            handleException(e);
+        }
+    }
+
+    private void doPutLink() {
+        try {
+            String linkName = JsonValueUtils.readString(full, "link_name");
+
+            ObjectStore os = nc.objectStore(bucket);
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            ObjectInfo oi = os.get(object, baos);
+            os.addLink(linkName, oi);
+            respond();
+        }
+        catch (Exception e) {
+            handleException(e);
+        }
+    }
+
+    private void doWatch() {
+        try {
+            ObjectStore os = nc.objectStore(bucket);
+            WatchWatcher ww = new WatchWatcher();
+            try (NatsObjectStoreWatchSubscription ws = os.watch(ww)) {
+                ww.latch.await(2, TimeUnit.SECONDS);
+            }
+            respond(ww.result);
+        }
+        catch (Exception e) {
+            handleException(e);
+        }
+    }
+
+    static class WatchWatcher implements ObjectStoreWatcher {
+        public CountDownLatch latch = new CountDownLatch(1);
+        public String result;
+
+        @Override
+        public void watch(ObjectInfo oi) {
+            if (result == null) {
+                result = oi.getDigest();
+            }
+            else {
+                result = result + "," + oi.getDigest();
+                latch.countDown();
+            }
+        }
+
+        @Override
+        public void endOfData() {}
+    }
+
+    private void doWatchUpdate() {
+        try {
+            ObjectStore os = nc.objectStore(bucket);
+            WatchUpdatesWatcher wuw = new WatchUpdatesWatcher();
+            try (NatsObjectStoreWatchSubscription ws = os.watch(wuw)) {
+                wuw.latch.await(2, TimeUnit.SECONDS);
+            }
+            respond(wuw.result);
+        }
+        catch (Exception e) {
+            handleException(e);
+        }
+    }
+
+    static class WatchUpdatesWatcher implements ObjectStoreWatcher {
+        public CountDownLatch latch = new CountDownLatch(1);
+        public String result;
+
+        @Override
+        public void watch(ObjectInfo oi) {
+            if (result == null) {
+                result = oi.getDigest();
+            }
+            else {
+                result = oi.getDigest();
+                latch.countDown();
+            }
+        }
+
+        @Override
+        public void endOfData() {}
+    }
+
+    protected void _respondDigest(ObjectInfo oi) {
```

Comment:
I don't think this validates what it's supposed to validate. You're taking digest from the object you got and just returning it back, so it will match whatever was created by the compatibility tester. What you should do is calculate the SHA for the object you got and return the SHA, not the digest from `ObjectInfo`.

Reply:
1. Will Fix.
2. We should just have a test that calculates a digest.

Reply:
As the client gets an object, it calculate the digest and checks that it matches what comes from the server. If the value does not match, I throw an exception.
```
if (totalBytes != oi.getSize()) { throw OsGetSizeMismatch.instance(); }
if (totalChunks != oi.getChunks()) { throw OsGetChunksMismatch.instance(); }
if (!digester.matches(oi.getDigest())) { throw OsGetDigestMismatch.instance(); }
```